### PR TITLE
feat: add LD proof to VP enclosing consent VC

### PR DESCRIPTION
**Pull request by llorllale**
part of #145 

Unblocks #191

Signed-off-by: George Aristy <george.aristy@securekey.com>

### CODE REVIEW

File: pkg/crypto/legacy_crypto.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	d, err := l.vdriReg.Resolve(signingDID)
+	if err != nil {
+		return "", fmt.Errorf("failed to resolve did : %w", err)
+	}
+
+	if len(d.PublicKey) == 0 {
+		return "", fmt.Errorf("no public keys found on did %s", signingDID)
+	}
+
+	return base58.Encode(d.PublicKey[0].Value), nil
```
Comment (rolsonquadras) [inline pkg/crypto/legacy_crypto.go:105]: Shouldn't the format of verificationMethod be like "did#keyID" ?

Comment (llorllale) [reply pkg/crypto/legacy_crypto.go:105]: @rolsonquadras It isn't (yet).

Comment (rolsonquadras) [reply pkg/crypto/legacy_crypto.go:105]: @llorllale I'm wondering how the receiver resolves the keys from DIDDoc and verify the signature ? Do we have any bdd tests where we are actually verifying the signature generated with this ?

Comment (llorllale) [reply pkg/crypto/legacy_crypto.go:105]: @rolsonquadras 

<details><summary>did:peer doc</summary>

```json
{                                                                                                                                                                                                                  
    "@context": [
        "https://w3id.org/did/v1"
    ],  
    "authentication": [
        ""
    ],  
    "created": "2020-07-24T17:08:31.370411561-04:00",
    "id": "did:peer:1zQmXJ88C99EXHjkpBEfaxKFeN9qnGadEZfQWs8SSH6rbVwy",
    "publicKey": [
        {
            "controller": "#id",
            "id": "", 
            "publicKeyBase58": "FMAHL95Quyt1WofUmKqWPYvjw3tnXKJ9MtWAmYfCg3M2",
            "type": "Ed25519VerificationKey2018"
        }
    ],  
    "service": [
        {
            "id": "#agent",
            "priority": 0,
            "recipientKeys": [
                "FMAHL95Quyt1WofUmKqWPYvjw3tnXKJ9MtWAmYfCg3M2"
            ],
            "routingKeys": null,
            "serviceEndpoint": "http://example.com/didcomm",
            "type": "did-communication"
        }
    ],  
    "updated": "2020-07-24T17:08:31.370411561-04:00"
}

```
</details>

<details><summary>signed VP</summary>

```json
{                                                                                                                                                                                                                  
    "@context": [
        "https://www.w3.org/2018/credentials/v1"
    ],  
    "proof": {
        "created": "2020-07-24T17:08:31.84389032-04:00",
        "jws": "eyJhbGciOiJFZERTQSIsImI2NCI6ZmFsc2UsImNyaXQiOlsiYjY0Il19..xy2NZy3TxBit6630-PkSuzxdRy4V1K8nq2WKX1-JG0SiLih1nXfY9mjNDCNhrE5xoz5LM7gkPojC6l7-tWR-Aw",
        "proofPurpose": "authentication",
        "type": "Ed25519Signature2018",
        "verificationMethod": "FMAHL95Quyt1WofUmKqWPYvjw3tnXKJ9MtWAmYfCg3M2"
    },  
    "type": "VerifiablePresentation",
    "verifiableCredential": [
        {
            "@context": [
                "https://www.w3.org/2018/credentials/v1",
                "https://www.w3.org/2018/credentials/examples/v1"
            ],
            "credentialSubject": {
                "degree": {
                    "degree": "MIT",
                    "type": "BachelorDegree"
                },
                "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
                "name": "Jayden Doe",
                "spouse": "did:example:c276e12ec21ebfeb1f712ebc6f1"
            },
            "id": "http://example.gov/credentials/ff98f978-588f-4eb0-b17b-60c18e1dac2c",
            "issuanceDate": "2020-03-16T22:37:26.544Z",
            "issuer": {
                "id": "did:web:vc.transmute.world",
                "name": "University"
            },
            "type": [
                "VerifiableCredential",
                "UniversityDegreeCredential"
            ]
        }
    ]   
}
```
</details>

The key is inlined in `VerificationMethod`.

Does the framework have an API to validate proofs? I assumed `verifiable.ParsePresentation()` did it.

**Comment by llorllale** [comment]
This is blocked while we investigate the framework's capability to validate a VP's embedded proof.

**Comment by llorllale** [comment]
hyperledger/aries-framework-go#2052 has been merged... this PR is now unblocked

**Comment by llorllale** [comment]
This PR is parked until other priorities get sorted out.

**Comment by llorllale** [comment]
Blocked again by the fact the framework's REST `/verifiable/presentation/generate` API does **not** use the legacy KMS ([code](https://github.com/hyperledger/aries-framework-go/blob/ddd20e12e74194177904cfd870c6117a7add11ba/pkg/controller/command/verifiable/command.go#L757-L791)). This API is needed to control the fixtures in the BDD tests. The legacy KMS is needed because the `vdri.Registry` is [still using the legacy KMS](https://github.com/hyperledger/aries-framework-go/blob/b972ae56a892f47348d1483dcfb5004ab688cf7f/pkg/vdri/registry.go#L107-L126) to create the DIDs (you can see the legacy KMS is always being created [here](https://github.com/hyperledger/aries-framework-go/blob/5c49f8159306d7cac93d65b79d2057138a1f4eaa/pkg/framework/aries/framework.go#L105-L147)).

I believe @Baha-sk is very close to pushing a PR that will delete the legacy KMS... we need to sync up.

**Comment by llorllale** [comment]
this was done in PR #308

**Pull request closed without merging**